### PR TITLE
RPG Analyzer

### PR DESCRIPTION
**Pull request by gcabbage**
This is a fix for [Thermo shell nanofac causes UI issues](https://ministry.kronosaur.com/record.hexm?id=71787) affecting analyzers.

When the nanofac OnUpdate event is called the dsRPGAnalyzeItem item list gets reset (see also [60413](https://ministry.kronosaur.com/record.hexm?id=60413)) and the `gItem` variable is changed to the nanofac causing various glitches as the dockscreen tries to use the nanofac to analyze items.

This PR saves the analyzer item as screen data which solves most of the UI problems (~~apart from reseting to the start of the list~~). Switching to a customItemPicker fixes the list reset

### CODE REVIEW

**Comment by gmoromisato** [comment]
I'd like to fix this in a different way. In particular, switching to customItemPicker seems wrong to me, since we'd have to switch all the ship dock screens to that. Instead, I'd like to fix the engine to preserve the list position.

Switching the item to screen data is a good change, since it avoids using global variables (always good) but I also want to fix the engine so that gItem does not get overwritten while you're in a dock screen (there is code to preserve it, but it fails in this particular case).

Bottom line, I'd like to take the "item as screen data" commit and the "analyzer destroys item" commit, but not the customItemPicker change.

I'll work on the engine changes this week.

**Comment by gcabbage** [comment]
I thought you'd prefer to fix it in the engine, but included the TLisp hack incase it was too far down the priority list. I'll rework the patch as requested (if I recall most of the "analyzer destroys item" commit is only needed due to the use of customItemPicker, but there is some tidying up that's worth saving)

BTW - the list reset issue only affects the case where the list comes from gSource  e.g.
* the sell items list (gSource=station player is docked at, but listing items from player) behaves correctly
* rpganalyzer and jettision screens (gSouce = player, listing items from player) is affected by reset

**Comment by gmoromisato** [comment]
Thank you!
